pkg/component/etcd/serviceaccount/serviceaccount_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceaccount_test
+
+import (
+	"context"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/pkg/client/kubernetes"
+	"github.com/gardener/etcd-druid/pkg/component/etcd/serviceaccount"
+
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("ServiceAccount Component", Ordered, func() {
+	var (
+		ctx    = context.TODO()
+		c      = fake.NewClientBuilder().WithScheme(kubernetes.Scheme).Build()
+		values = &serviceaccount.Values{
+			Name:      "test-service-account",
+			Namespace: "test-namespace",
+			Labels: map[string]string{
+				"foo": "bar",
+			},
+			DisableAutomount: true,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         druidv1alpha1.GroupVersion.String(),
+					Kind:               "Etcd",
+					Name:               "test-etcd",
+					UID:                "123-456-789",
+					Controller:         pointer.Bool(true),
+					BlockOwnerDeletion: pointer.Bool(true),
+				},
+			},
+		}
+		saComponent = serviceaccount.New(c, values)
+	)
+
+	Context("#Deploy", func() {
+		It("should create the ServiceAccount with the expected values", func() {
+			By("creating a ServiceAccount")
+			err := saComponent.Deploy(ctx)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("verifying that the ServiceAccount is created on the K8s cluster as expected")
+			created := &corev1.ServiceAccount{}
+			err = c.Get(ctx, getServiceAccountKeyFromValue(values), created)
+			Expect(err).NotTo(HaveOccurred())
+			verifyServicAccountValues(created, values)
+		})
+		It("should update the ServiceAccount with the expected values", func() {
```

Comment:
You assume that the SA is created in the previous test. If the tests are run concurrently then this would not work. Each test should ideally be independent of each other. If you have to establish deterministic execution order for tests then you should explicitly use https://onsi.github.io/ginkgo/#ordered-containers